File: interface-pc/beehoneyt/mesureruche.cpp
#include "communication.h"
#include "mesureruche.h"
#include "ihmpc.h"
#include <QDebug>
#include <QObject>

/**
* @file mesureruche.cpp
* @brief La définition de la classe MesureRuche
* $LastChangedRevision: 116 $
* $LastChangedDate: 2021-05-26 10:40:13 +0200 (mer. 26 mai 2021) $
*/

/**
 * @brief Constructeur de la classe MesureRuche
 * @fn MesureRuche::MesureRuche
 * @param parent
 * @param temperatureInterieure
 * @param temperatureExterieure
 * @param humiditeInterieure
 * @param humiditeExterieure
 * @param pression
 * @param poids
 */
MesureRuche::MesureRuche(QObject *parent, double temperatureInterieure , double temperatureExterieure , double humiditeInterieure , double humiditeExterieure,
                         double pression , double poids) : QObject(parent),
                                                           temperatureInterieure(temperatureInterieure),
                                                           temperatureExterieure(temperatureExterieure),
                                                           humiditeInterieure(humiditeInterieure),
                                                           humiditeExterieure(humiditeExterieure),
                                                           pression(pression),
                                                           poids(poids)
{
    qDebug() << Q_FUNC_INFO;
}

/**
 * @brief Constructeur de la classe MesureRuche par recopie
 * @fn MesureRuche::MesureRuche
 * @param mesureRuche
 */
MesureRuche::MesureRuche(const MesureRuche &mesureRuche) : QObject(mesureRuche.parent()),
                                                           temperatureInterieure(mesureRuche.temperatureInterieure),
                                                           temperatureExterieure(mesureRuche.temperatureExterieure),
                                                           humiditeInterieure(mesureRuche.humiditeInterieure),
                                                           humiditeExterieure(mesureRuche.humiditeExterieure),
                                                           pression(mesureRuche.pression),
                                                           poids(mesureRuche.poids),
                                                           horodatage(mesureRuche.horodatage)
{
    qDebug() << Q_FUNC_INFO;
}

/**
 * @brief Destructeur de la classe MesureRuche
 * @fn MesureRuche::~MesureRuche
 */
MesureRuche::~MesureRuche()
{
    qDebug() << Q_FUNC_INFO;
}

// Services ==============================================================
/**
 * @brief Methode qui intialise les attributs
 * @fn MesureRuche::initialiser
 * @return void
 */
void MesureRuche::initialiser()
{
    temperatureInterieure = 0.;
    temperatureExterieure = 0.;
    humiditeInterieure = 0.;
    humiditeExterieure = 0.;
    pression = 0.;
    poids = 0.;
    horodatage = "";
}

// Mutateurs =============================================================
/**
 * @brief Methode qui change la valeur de la temperature interieure
 * @fn MesureRuche::setTemperatureInterieure
 * @param temperatureInterieure
 * @return void
 */
void MesureRuche::setTemperatureInterieure(double temperatureInterieure)
{
    this->temperatureInterieure=temperatureInterieure;
}

/**
 * @brief Methode qui change la valeur de la temperature exterieure
 * @fn MesureRuche::setTemperatureExterieure
 * @param temperatureExterieure
 * @return void
 */
void MesureRuche::setTemperatureExterieure(double temperatureExterieure)
{
    this->temperatureExterieure=temperatureExterieure;
}

/**
 * @brief Methode qui change la valeur de l'humidite interieure
 * @fn MesureRuche::setHumiditeInterieure
 * @param humiditeInterieure
 * @return void
 */
void MesureRuche::setHumiditeInterieure(double humiditeInterieure)
{
    this->humiditeInterieure=humiditeInterieure;
}

/**
 * @brief Methode qui change la valeur de l'humidite exterieure
 * @fn MesureRuche::setHumiditeExterieure
 * @param humiditeExterieure
 * @return void
 */
void MesureRuche::setHumiditeExterieure(double humiditeExterieure)
{
    this->humiditeExterieure=humiditeExterieure;
}

/**
 * @brief Methode qui change la valeur du poids
 * @fn MesureRuche::setPoids
 * @param poids
 * @return void
 */
void MesureRuche::setPoids(double poids)
{
    this->poids=poids;
}

/**
 * @brief Methode qui change la valeur de la pression
 * @fn MesureRuche::setPression
 * @param pression
 * @return void
 */
void MesureRuche::setPression(double pression)
{
    this->pression=pression;
}

/**
 * @brief Methode qui change la valeur de l'horodatage
 * @fn MesureRuche::setHorodatage
 * @param horodatage
 */
void MesureRuche::setHorodatage(QString horodatage)
{
    this->horodatage=horodatage;
}

// Accesseurs ============================================================
/**
 * @brief Methode qui retourne la valeur de la temperature interieure
 * @fn MesureRuche::getTemperatureInterieure
 * @return double
 */
double MesureRuche::getTemperatureInterieure() const
{
    return this->temperatureInterieure;
}

/**
 * @brief Methode qui retourne la valeur de la temperature exterieure
 * @fn MesureRuche::getTemperatureExterieure
 * @return double
 */
double MesureRuche::getTemperatureExterieure() const
{
    return this->temperatureExterieure;
}

/**
 * @brief Methode qui retourne la valeur de l'humidite interieure
 * @fn MesureRuche::getHumiditeInterieure
 * @return double
 */
double MesureRuche::getHumiditeInterieure() const
{
    return this->humiditeInterieure;
}

/**
 * @brief Methode qui retourne la valeur de l'humidite exterieure
 * @fn MesureRuche::getHumiditeExterieure
 * @return
 */
double MesureRuche::getHumiditeExterieure() const
{
    return this->humiditeExterieure;
}

/**
 * @brief Methode qui retourne la valeur du poids
 * @fn MesureRuche::getPoids
 * @return double
 */
double MesureRuche::getPoids() const
{
    return this->poids;
}

/**
 * @brief Methode qui retourne la valeur de la pression
 * @fn MesureRuche::getPression
 * @return double
 */
double MesureRuche::getPression() const
{
    return this->pression;
}

/**
 * @brief Methode qui retourne la valeur de l'horodatage
 * @fn MesureRuche::getHorodatage
 * @return QString
 */
QString MesureRuche::getHorodatage() const
{
    return this->horodatage;
}

File: interface-android/BeeHoneyt/app/src/main/java/com/example/bee_honeyt/MesureRuche.java
package com.example.bee_honeyt;

/**
 * @file MesureRuche.java
 * @brief Déclaration de la classe MesureRuche
 * @details Contient les mesures d’une ruche sélectionnée.
 * @author FOUCAULT Clémentine
 * $LastChangedRevision: 129 $
 * $LastChangedDate: 2021-05-28 16:20:17 +0200 (ven. 28 mai 2021) $
 */

/**
 * @class MesureRuche
 * @brief Les mesures associées à une ruche
 */
public class MesureRuche
{
    private static final String TAG = "_MesureRuche"; //!< TAG pour les logs

    private double temperatureInterieure = 0.0; //!< La température interieure
    private double temperatureExterieure= 0.0; //!< La température exterieure
    private int humiditeInterieure = 0; //!< L'humidité interieure
    private int humiditeExterieure = 0; //!< L'humidité exterieure
    private int pression = 0; //!< La pression
    private double poids = 0.0; // le poids

    /**
     * @brief constructeur par defaut
     */
    public MesureRuche()
    {

    }

    /**
     * @brief constructeur de la classe MesureRuche
     */
    public MesureRuche(double temperatureInterieure, double temperatureExterieure, int humiditeInterieure, int humiditeExterieure, int pression, double poids)
    {
        this.temperatureInterieure = temperatureInterieure;
        this.temperatureExterieure = temperatureExterieure;
        this.humiditeInterieure = humiditeInterieure;
        this.humiditeExterieure = humiditeExterieure;
        this.pression = pression;
        this.poids = poids;

    }

    /**
     * @brief accesseur Temperature interieure
     * @return double la temperature interieure
     */
    public final double getTemperatureInterieure()
    {
        return temperatureInterieure;
    }

    /**
     * @brief mutateur Temperature interieure
     * @param temperatureInterieure
     */
    public void setTemperatureInterieure(double temperatureInterieure)
    {
        this.temperatureInterieure = temperatureInterieure;
    }

    /**
     * @brief accesseur Temperature exterieure
     * @return double la temperature exterieure
     */
    public final double getTemperatureExterieure()
    {
        return temperatureExterieure;
    }

    /**
     * @brief mutateur Temperature exterieure
     * @param temperatureExterieure
     */
    public void setTemperatureExterieure(double temperatureExterieure)
    {
        this.temperatureExterieure = temperatureExterieure;
    }

    /**
     * @brief accesseur Humidite interieure
     * @return int l'humidite interieure
     */
    public final int getHumiditeInterieure()
    {
        return humiditeInterieure;
    }

    /**
     * @brief mutateur Humidite interieure
     * @param humiditeInterieure
     */
    public void setHumiditeInterieure(int humiditeInterieure)
    {
        this.humiditeInterieure = humiditeInterieure;
    }

    /**
     * @brief accesseur Humidite exterieure
     * @return int l'humidite exterieure
     */
    public final int getHumiditeExterieure()
    {
        return humiditeExterieure;
    }

    /**
     * @brief mutateur Humidite exterieure
     * @param humiditeExterieure
     */
    public void setHumiditeExterieure(int humiditeExterieure)
    {
        this.humiditeExterieure = humiditeExterieure;
    }

    /**
     * @brief accesseur Pression
     * @return int la pression
     */
    public final int getPression()
    {
        return pression;
    }

    /**
     * @brief mutateur Pression
     * @param pression
     */
    public void setPression(int pression)
    {
        this.pression = pression;
    }

    /**
     * @brief accesseur Poids
     * @return double le poids
     */
    public final double getPoids()
    {
        return poids;
    }

    /**
     * @brief mutateur Poids
     * @param poids
     */
    public void setPoids(double poids)
    {
        this.poids = poids;
    }
}
